src/api/app/models/kiwi/repository.rb

```diff
@@ -50,7 +53,31 @@ def source_path_format
     errors.add(:source_path, "has an invalid format")
   end
 
+  def to_xml
+    repo_attributes = { type: repo_type }
+    repo_attributes[:status] = 'replaceable' if replaceable
+    repo_attributes[:priority] = priority if priority.present?
+    repo_attributes[:alias] = self.alias if self.alias.present?
+    if username.present?
+      repo_attributes[:username] = username
+      repo_attributes[:password] = password
+    end
+
+    builder = Nokogiri::XML::Builder.new
+    builder.repository(repo_attributes) do |repo|
+      repo.source(path: source_path)
+    end
+
+    builder.to_xml save_with: Nokogiri::XML::Node::SaveOptions::NO_DECLARATION | Nokogiri::XML::Node::SaveOptions::FORMAT
+  end
+
   #### Alias of methods
+
+  private
+
+  def not_outdated
+    errors.add(:base, 'Image configuration has changed') if image.package.kiwi_image_outdated?
```

Comment:
btw, the kiwi_image_outdated? method is missing tests:-) I know it's not added here though